drivers/ethernet/eth_stm32_hal.c

```diff
@@ -864,6 +875,8 @@ static const struct eth_stm32_hal_dev_cfg eth0_config = {
 	.pclken_ptp = {.bus = DT_INST_CLOCKS_CELL_BY_NAME(0, mac_clk_ptp, bus),
 		       .enr = DT_INST_CLOCKS_CELL_BY_NAME(0, mac_clk_ptp, bits)},
 #endif /* !CONFIG_SOC_SERIES_STM32H7X */
+	.pinctrl = eth0_pins,
+	.pinctrl_len = ARRAY_SIZE(eth0_pins),
```

Comment:
Maybe the following would be more in line with driver style:
```
	.pinctrl = {
		.list = eth0_pins,
		.len = ARRAY_SIZE(eth0_pins)
	}
```

Reply:
I think it's better to align with other STM32 drivers. In this driver multiple sub-fields are initialized because the structures are given this way. Maybe pinctrl driver can provide such struct (similar to `struct stm32_pclken`) plus an initializer macro.

Reply:
Agreed.